mutation_reader.hh

```diff
@@ -136,6 +136,8 @@ flat_mutation_reader make_filtering_reader(flat_mutation_reader rd, MutationFilt
     return make_flat_mutation_reader<filtering_reader<MutationFilter>>(std::move(rd), std::forward<MutationFilter>(filter));
 }
 
+flat_mutation_reader make_filtering_reader(flat_mutation_reader, const dht::partition_range&, const query::partition_slice&);
```

Comment:
Please add a comment to explain what this is doing.

Reply:
not part of this PR, explained below